test/common/config.js

```diff
@@ -56,6 +56,10 @@ export default function () {
         ...(!!process.env.CODE_COVERAGE
           ? [`--plugin-path=${path.join(__dirname, 'fixtures', 'plugins', 'coverage')}`]
           : []),
+        '--logging.appenders.deprecation.type=console',
+        '--logging.appenders.deprecation.layout.type=json',
+        '--logging.loggers[0].name=elasticsearch.deprecation',
+        '--logging.loggers[0].appenders[0]=deprecation',
```

Comment:
Deprecation messages contain newlines and then tooling_log_text_writer prints only the first line but ignores subsequent lines with the query and stack trace. Having this in JSON isn't ideal but it's the best I can think of 🤷